### PR TITLE
Add more debuggable information for the Display already acquired exception

### PR DESCRIPTION
**Pull request by Amejia481**
---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt [hunk 1]
```diff
@@ -38,9 +38,14 @@ class GeckoEngineView @JvmOverloads constructor(
             } catch (e: IllegalStateException) {
                 // This is to debug "display already acquired" crashes
                 val otherActivityClassName =
-                    this.session?.accessibility?.view?.context?.javaClass?.simpleName
+                        this.session?.accessibility?.view?.context?.javaClass?.simpleName
+                val otherActivityClassHashcode =
+                        this.session?.accessibility?.view?.context?.javaClass?.hashCode()
                 val activityClassName = context.javaClass.simpleName
-                val msg = "ATTACH VIEW: Current activity: $activityClassName Other activity: $otherActivityClassName"
+                val activityClassHashCode = context.javaClass.hashCode()
```
Comment (Amejia481) [inline components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt:45]: @pocmo is `context.javaClass.hashCode()`  the right `hashCode` ? I'm a bit confused if this is the `hashCode` for the `Class` object or if this is the one for the activity object?

Comment (pocmo) [reply components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt:45]: Oh, I missed that. No, we want the hashCode of the context instances here! :)

Comment (Amejia481) [reply components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt:45]: Thanks :)

**Comment by Amejia481** [comment]
bors r=pocmo

**Comment by csadilek** [comment]
@Amejia481 @pocmo There are two cases where we're handling this exception. Let's add the hash codes here as well? https://github.com/mozilla-mobile/android-components/blob/master/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt#L118

Although the current one does cover our top crash.

**Comment by Amejia481** [comment]
It's not a problem. Let add the has for that case too!

**Pull request merged**